src/main/java/configurationslicing/promotedbuilds/PromotedBuildsNameSlicer.java

```diff
@@ -0,0 +1,155 @@
+package configurationslicing.promotedbuilds;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.base.Throwables;
+import configurationslicing.UnorderedStringSlicer;
+import hudson.Extension;
+import hudson.matrix.MatrixProject;
+import hudson.model.AbstractProject;
+import hudson.model.Project;
+import hudson.plugins.promoted_builds.JobPropertyImpl;
+import hudson.plugins.promoted_builds.PromotionProcess;
+import jenkins.model.Jenkins;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Extension
+public class PromotedBuildsNameSlicer extends UnorderedStringSlicer<AbstractProject<?,?>> {
+
+
+    public PromotedBuildsNameSlicer() {super(new PromotedBuildsSlicerSpec()); }
+
+    public static class PromotedBuildsSlicerSpec extends UnorderedStringSlicerSpec<AbstractProject<?,?>> {
+
+        private static final Logger log = LoggerFactory.getLogger(PromotedBuildsSlicerSpec.class);
+        public static final String NOTHING = "(NOTHING)";
+
+        @Override
+        public String getName() { return "Promoted Builds Names"; }
+
+        @Override
+        public String getUrl() { return "promotedbuilds"; }
+
+        @Override
+        public String getDefaultValueString() { return ""; }
```

Comment:
Added a comment. To expand on it for reference: using "(Default)" as the default string here could imply that there is a default value set but this field has no concept of a default value so we return "" in an attempt to not be misleading.